src/MOI_wrapper.jl

```diff
@@ -450,27 +450,37 @@ function MOI.delete(model::Optimizer, v::MOI.VariableIndex)
         end
     end
     model.name_to_variable = nothing
+    # We throw away name_to_constraint_index so we will rebuild SingleVariable
+    # constraint names without v.
+    model.name_to_constraint_index = nothing
     return
 end
 
 function MOI.get(model::Optimizer, ::Type{MOI.VariableIndex}, name::String)
     if model.name_to_variable === nothing
         _rebuild_name_to_variable(model)
     end
-    return get(model.name_to_variable, name, nothing)
+    if haskey(model.name_to_variable, name)
+        variable = model.name_to_variable[name]
+        if variable === nothing
+            error("Duplicate variable name detected: $(name)")
+        end
+        return variable
+    end
+    return nothing
 end
 
 function _rebuild_name_to_variable(model::Optimizer)
-    model.name_to_variable = Dict{String, MOI.VariableIndex}()
+    model.name_to_variable = Dict{String, Union{Nothing, MOI.VariableIndex}}()
     for (index, info) in model.variable_info
         if info.name == ""
             continue
         end
         if haskey(model.name_to_variable, info.name)
-            model.name_to_variable = nothing
-            error("Duplicate variable name detected: $(info.name)")
+            model.name_to_variable[info.name] = nothing
```

Comment:
We use `MOI.VariableIndex(0)` in `MOIU.Model`

Reply:
I'd prefer to keep `nothing`. It's more explicit, and the small union shouldn't be a performance bottleneck.